open-craft/xblock-poll
# MCKIN-8752: Complete poll xblock when response submitted.

Changes completion mode from complete-on-view to complete-on-submit.

**JIRA tickets**: MCKIN-8752

**Discussions**: N/A

**Dependencies**: None

**Screenshots**:

**Sandbox URL**: TBD - sandbox is being provisioned.

**Merge deadline**: ASAP

**Testing instructions**:

1. Create a poll xblock and a survey xblock.
2. Visit them.
3. Verify that no BlockCompletion objects are created.
4. Submit responses.
5. Verify that BlockCompletion objects are created.

**Author notes and concerns**:

test_submit_button is flaky.  the submit button click handler disables the button, and re-enables it before it is done.  The test tries to sneak in while it is disabled, and check that it's disabled, but it is frequently re-enabled too quickly for the test to catch it.  Verified manually that the button disables briefly when submitting a response, but I can't spend hours right now re-running the tests to get the two test runs to match.

**Reviewers**

**Settings**
```yaml
EDXAPP_FEATURES:
  ENABLE_COMBINED_LOGIN_REGISTRATION: true
```
